File: app/src/main/java/com/pengxh/autodingding/utils/EmailAuthenticator.java
package com.pengxh.autodingding.utils;

import javax.mail.Authenticator;
import javax.mail.PasswordAuthentication;


public class EmailAuthenticator extends Authenticator {

    private final String userName;
    private final String password;

    public EmailAuthenticator(String username, String password) {
        this.userName = username;
        this.password = password;
    }

    protected PasswordAuthentication getPasswordAuthentication() {
        return new PasswordAuthentication(userName, password);
    }
}

File: app/src/main/java/com/pengxh/autodingding/utils/EmailAuthenticator.kt
package com.pengxh.autodingding.utils

import javax.mail.Authenticator
import javax.mail.PasswordAuthentication

class EmailAuthenticator(private val userName: String, private val password: String) :
    Authenticator() {
    override fun getPasswordAuthentication(): PasswordAuthentication {
        return PasswordAuthentication(userName, password)
    }
}